packages/components/src/custom-select-control-v2/use-deprecated-props.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CustomSelectItem } from '.';
+import type { CustomSelectProps, LegacyCustomSelectProps } from './types';
+
+function isLegacyProps( props: any ): props is LegacyCustomSelectProps {
+	return (
+		typeof props.options !== 'undefined' ||
+		props.__experimentalShowSelectedHint !== undefined
+	);
+}
+
+const transformOptionsToChildren = ( props: LegacyCustomSelectProps ) => {
+	if ( props.options === undefined ) {
+		return;
+	}
+
+	return props.options.map(
+		( { name, key, __experimentalHint, ...rest }: any ) => {
+			const withHint = (
+				<>
+					<span>{ name }</span>
+					<span className="components-custom-select-control__item-hint">
+						{ __experimentalHint }
+					</span>
+				</>
+			);
+
+			return (
+				<CustomSelectItem
+					{ ...rest }
+					key={ key }
+					value={ name }
+					children={
+						props.__experimentalShowSelectedHint ? withHint : name
```

Comment:
I am wondering if we need to support the experimental hint. Since it wasn't intended to be a permanent feature and it hasn't been included in the README.